.github/workflows/main.yml

```diff
@@ -11,9 +11,9 @@ jobs:
         python-version: ["3.5", "3.6", "3.7", "3.8", "3.9", "3.10", "pypy3"]
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
```

Comment:
In order to support Python 3.5 and 3.6 with `actions/setup-python@v4`, you will need to downgrade to `ubuntu-20.04`.

Reply:
It was `v2` when it first failed.